blockchain/chain_sync/src/sync.rs

```diff
@@ -542,8 +797,339 @@ where
+        });
+    }
+
+    #[test]
+    fn inform_new_head_given_fts_test() {
+        let fts = construct_full_tipset();
+        let source = PeerId::random();
+
+        let db = MemoryDB::default();
+        let mut cs = chain_syncer_setup(Arc::new(db));
+
+        task::spawn(async move {
+            cs.inform_new_head(&source, &fts)
+                .await
+                .expect("error with informing of new head");
+        });
+    }
+
+    #[test]
+    fn set_peer_head_and_select_target_test() {
+        let ts = construct_tipset(1);
+        let source = PeerId::random();
+
+        let db = MemoryDB::default();
+        let mut cs = chain_syncer_setup(Arc::new(db));
+
+        task::spawn(async move {
+            cs.set_peer_head(&source, ts.clone())
+                .await
+                .expect("error setting peer head");
+            for (peer_id, tip) in cs.peer_manager.peer_heads.read().await.iter() {
+                assert_eq!(peer_id, &source);
+                assert_eq!(tip, &ts.clone());
+            }
+            assert_eq!(cs.select_sync_target().await.unwrap(), ts);
+        });
+    }
+
+    #[test]
+    fn schedule_tipset_given_tipset_test() {
+        let ts = construct_tipset(1);
+        let ts_copy = construct_tipset(1);
+
+        let db = MemoryDB::default();
+        let mut cs = chain_syncer_setup(Arc::new(db));
+
+        task::spawn(async move {
+            cs.schedule_tipset(ts.clone())
+                .await
+                .expect("error scheduling tipsets");
+            assert_eq!(cs.next_sync_target._tipsets(), &[ts]);
+
+            cs.schedule_tipset(ts_copy)
+                .await
+                .expect("error scheduling tipsets");
+            assert_eq!(cs.next_sync_target._tipsets().len(), 2);
+        });
+    }
+
+    #[test]
+    fn compute_msg_data_given_msgs_test() {
+        let (bls, secp) = construct_messages();
+
+        let db = MemoryDB::default();
+        let cs = chain_syncer_setup(Arc::new(db));
+        let expected_root =
+            Cid::from_raw_cid("bafy2bzaced5inutkibck2wagtnggbvjpbr65ghdncivs3gpagx67s3xs3i5wa")
+                .unwrap();
+
+        let root = cs.compute_msg_data(&[bls], &[secp]).unwrap();
+        assert_eq!(root, expected_root);
+    }
+
+    #[test]
+    fn validate_msg_data_given_msgs_test() {
+        let block = construct_block();
+
+        let db = MemoryDB::default();
+        let cs = chain_syncer_setup(Arc::new(db));
+
+        cs.validate_msg_data(&block).expect("invalid msg data");
+    }
+
+    #[test]
+    fn tipset_msgs_given_tsb_test() {
+        let db = MemoryDB::default();
+        let cs = chain_syncer_setup(Arc::new(db));
+
+        let ts = construct_tipset(1);
+        let (bls, secp) = construct_messages();
+
+        let fts: FullTipset = cs
+            .tipset_msgs(
+                ts.clone(),
+                &[bls],
+                &[secp],
+                ts.blocks().len(),
+                ts.blocks().len(),
```

Comment:
This is supposed to be the bls and msgs lengths, which I don't see why that should ever be parameters, because if that was the length that it was supposed to actually be, you have the two slices of bls and secp messages, length is redundant. You just created this test to follow the logic of the function, I don't think this test is very good

Reply:
Will remove.